lib/MusicBrainz/Server/WebService/Serializer/JSON/2/Role/Relationships.pm

```diff
@@ -11,11 +11,16 @@ around serialize => sub
     my ($orig, $self, $entity, $inc, $opts, $toplevel) = @_;
     my $ret = $self->$orig($entity, $inc, $opts, $toplevel);
 
-    return $ret unless defined $inc && $inc->has_rels;
+    return $ret unless
+        defined $inc &&
+        $inc->has_rels &&
+        $entity->are_relationships_loaded;
 
     my @rels = map { serialize_entity($_, $inc, $opts) }
-        sort_by { $_->target_key . $_->link->type->name }
-            @{ $entity->relationships };
+        sort_by { $_->link->type->name }
+        sort_by { $_->link_order // 0 }
+        sort_by { $_->target_key }
```

Comment:
Yes, but the sort isn’t stable. At least it’s not guaranteed to be.

The way it was done previously (i.e. concatenating the keys) should be safe, in particular when using a separator that sorts less than all characters that can appear in the parts (TAB should work). E.g.

    sort_by { join("\t", $_->target_key, (sprintf "%09d", $_->link_order // 0), $_->link->type->name) ) }